src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/NewHandler.java

```diff
@@ -39,33 +42,81 @@ public NewHandler(IndentationCheck indentCheck,
                       DetailAST ast,
                       AbstractExpressionHandler parent) {
         super(indentCheck, "operator new", ast, parent);
+        mainAst = ast;
     }
 
     @Override
     public void checkIndentation() {
-        final DetailAST type = getMainAst().getFirstChild();
-        if (type != null) {
-            checkExpressionSubtree(type, getIndent(), false, false);
+        // if new is on the line start and it is not the part of assignment.
+        if (isOnStartOfLine(mainAst)
+                && !isNewKeywordFollowedByAssign()) {
+            final int columnNo = expandedTabsColumnNo(mainAst);
+            final IndentLevel level = getIndentImpl();
+
+            if (columnNo < level.getFirstIndentLevel()) {
+                logError(mainAst, "", columnNo, level);
+            }
+        }
+
+        final DetailAST firstChild = mainAst.getFirstChild();
+        if (firstChild != null) {
+            checkExpressionSubtree(firstChild, getIndent(), false, false);
         }
 
-        final DetailAST lparen = getMainAst().findFirstToken(TokenTypes.LPAREN);
+        final DetailAST lparen = mainAst.findFirstToken(TokenTypes.LPAREN);
         checkLeftParen(lparen);
     }
 
+    @Override
+    public IndentLevel getSuggestedChildIndent(AbstractExpressionHandler child) {
+        final int offset;
+        if (child.getMainAst().getType() == TokenTypes.OBJBLOCK) {
```

Comment:
This is a good candidate for `TokenUtil.isOfType()`.

Reply:
Done

Reply:
Resolved.